types/src/transaction/transaction.rs

```diff
@@ -264,6 +232,148 @@ fn is_input_and_output_consistent(inputs: &[AssetTransferInput], outputs: &[Asse
     sum.iter().all(|(_, sum)| sum.is_zero())
 }
 
+fn apply_bitmask(
+    mut bitmask: u128,
+    outputs: Vec<AssetTransferOutput>,
+    mut result: Vec<AssetTransferOutput>,
+) -> Result<Vec<AssetTransferOutput>, HashingError> {
+    for i in 0.. {
+        if bitmask == 0 {
+            return Ok(result)
+        }
+
+        if outputs.is_empty() {
+            return Err(HashingError::InvalidFilter)
+        }
+
+        let mut output = outputs[i].clone();
```

Comment:
1. No reason to make `output` mutable.
2. No reason to clone `outputs[i]` if it's not pushed in the result.

```rust
if (bitmask & 0x1) == 1 {
    result.push(outputs[i].clone());
}
```

Reply:
I resolved it